### PR TITLE
Features/5694 upgrade test container

### PR DESCRIPTION
**Pull request by utas-raymondng**

### CODE REVIEW

File: indexer/src/main/java/au/org/aodn/esindexer/configuration/GeoNetworkSearchConfig.java [hunk 1]
```diff
@@ -17,16 +18,37 @@
...
 @Configuration
 public class GeoNetworkSearchConfig {
 
     @Bean(name = "gn4ElasticsearchClient")
-    public ElasticsearchClient createGN4ElasticsearchClient(@Qualifier("gn4ElasticRestClient") RestClient restClient) {
+    public ElasticsearchClient createGN4ElasticsearchClient(@Qualifier("gn4ElasticRestClient") RestClient restClient) throws NoSuchFieldException, IllegalAccessException {
 
         RestClientTransport c = new RestClientTransport(restClient, new JacksonJsonpMapper());
+        // Fix due to elastic search api update
+        // Need a hack to work around an issue with geonetwork, geonetwork exposed the ElasticSearch api via endpoint _search
```
Comment (vietnguyengit) [inline indexer/src/main/java/au/org/aodn/esindexer/configuration/GeoNetworkSearchConfig.java:32]: can you add a TODO for easier to track in future?

File: indexer/src/main/java/au/org/aodn/esindexer/utils/MapperUtils.java [hunk 2]
```diff
@@ -59,15 +59,20 @@ public static String mapContactsPosition(CIIndividualPropertyType2 individual) {
      */
     public static ContactsAddressModel mapContactsAddress(CIAddressPropertyType2 address) {
         ContactsAddressModel addressItem = ContactsAddressModel.builder().build();
-        List<String> deliveryPoints = new ArrayList<>();
 
-        address.getCIAddress().getDeliveryPoint().forEach(deliveryPoint -> {
-            String deliveryPointString = deliveryPoint.getCharacterString().getValue().toString();
-            deliveryPoints.add(deliveryPointString != null ? deliveryPointString : "");
-        });
+        List<au.org.aodn.metadata.iso19115_3_2018.CharacterStringPropertyType> v = safeGet(() -> address.getCIAddress().getDeliveryPoint());
```
Comment (vietnguyengit) [inline indexer/src/main/java/au/org/aodn/esindexer/utils/MapperUtils.java:63]: what does `v` stand for?

Comment (vietnguyengit) [reply indexer/src/main/java/au/org/aodn/esindexer/utils/MapperUtils.java:63]: can you rename it with something else easier to read?

File: indexer/src/test/java/au/org/aodn/esindexer/BaseTestClass.java [hunk 3]
```diff
@@ -187,31 +189,51 @@ public void login() {
...
-        ResponseEntity<String> r = testRestTemplate
-                .exchange(
-                        getRecordUrl(uuid),
-                        HttpMethod.DELETE,
-                        requestEntity,
-                        String.class
-                );
-        logger.info("{}", r.getStatusCode());
+            // Index the item so that query yield the right result before delete
+            ResponseEntity<Void> t = testRestTemplate
```
Comment (vietnguyengit) [inline indexer/src/test/java/au/org/aodn/esindexer/BaseTestClass.java:199]: same here, `t`

File: indexer/src/test/java/au/org/aodn/esindexer/BaseTestClass.java [hunk 4]
```diff
@@ -187,31 +189,51 @@ public void login() {
...
-                .exchange(
-                        getIndexUrl(),
-                        HttpMethod.PUT,
-                        requestEntity,
-                        Void.class
-                );
-        assertEquals("Trigger index OK", HttpStatus.OK, t.getStatusCode());
+            for(String uuid: uuids) {
+                logger.info("Deleting GN doc {}", uuid);
+                ResponseEntity<String> r = testRestTemplate
```
Comment (vietnguyengit) [inline indexer/src/test/java/au/org/aodn/esindexer/BaseTestClass.java:212]: and `r` here as well , these single letter variable names growing, will get to the point `t` in this class means this, `t` in that class means that ... etc. maybe hard to read and understand in future.

**Review by vietnguyengit — COMMENTED** [review]
LGTM, had some comments on variable naming, it's up to you to change or not.

File: indexer/src/main/java/au/org/aodn/esindexer/configuration/GeoNetworkSearchConfig.java (hunk 1 above)
Comment (vietnguyengit) [reply indexer/src/main/java/au/org/aodn/esindexer/configuration/GeoNetworkSearchConfig.java:32]: if there will be something to do later for getting away from work around strategy

File: indexer/src/main/java/au/org/aodn/esindexer/utils/MapperUtils.java (hunk 2 above)
Comment (utas-raymondng) [reply indexer/src/main/java/au/org/aodn/esindexer/utils/MapperUtils.java:63]: changed

File: indexer/src/test/java/au/org/aodn/esindexer/BaseTestClass.java (hunk 3 above)
Comment (utas-raymondng) [reply indexer/src/test/java/au/org/aodn/esindexer/BaseTestClass.java:199]: done

File: indexer/src/test/java/au/org/aodn/esindexer/BaseTestClass.java (hunk 4 above)
Comment (utas-raymondng) [reply indexer/src/test/java/au/org/aodn/esindexer/BaseTestClass.java:212]: done

**Review by vietnguyengit — APPROVED** [review]
LGTM , thank you.

**Pull request merged**